File: alura/carreira_java/example26/src/br/com/flaviogf/Employee.java
package br.com.flaviogf;

public abstract class Employee {
    private String name;
    private String document;
    private double salary;

    public Employee(String name, String document, double salary) {
        this.name = name;
        this.document = document;
        this.salary = salary;
    }

    public String getName() {
        return name;
    }

    public String getDocument() {
        return document;
    }

    public double getSalary() {
        return salary;
    }

    public abstract double getBonus();
}

File: alura/orientacao_a_objetos_com_php_002/bank/src/Models/Employee.php
<?php

namespace Bank\Models;

abstract class Employee extends Person
{
    private float $salary;

    public function __construct(string $name, Document $document, float $salary)
    {
        parent::__construct($name, $document);
        $this->salary = $salary;
    }

    public function getSalary(): float
    {
        return $this->salary;
    }

    abstract public function getBonus(): float;
}